lib/plenty_client/request.rb

```diff
@@ -110,13 +110,13 @@ def base_url(path)
       # 2017-12-04 DO: there has to be a supervisor watching over the users limits
       #                BEFORE the request actually happens
       #                response_header is after the request and useless if you have multiple instances of the Client
-      def throttle_check_short_period(response_header)
+      def throttle_check_short_period(response_header, bandwidth = 1)
         short_calls_left = response_header['X-Plenty-Global-Short-Period-Calls-Left']
         short_seconds_left = response_header['X-Plenty-Global-Short-Period-Decay']
         return if short_calls_left&.empty? || short_seconds_left&.empty?
         return if short_calls_left.to_i > 10 && short_seconds_left.to_i > 3
 
-        PlentyClient::Config.request_wait_until = Time.now + short_seconds_left.to_i
+        PlentyClient::Config.request_wait_until = Time.now + (short_seconds_left + 2).to_i
```

Comment:
what about short_seconds_left from 3 into 5?

Reply:
The problem is not that is stopping too late @owen2345, but rather that it is starting back again too early sometimes. 

i.e. 
👉 `short_seconds_left = 10.30` seconds.
👉 `short_seconds_left.to_i` will result in 10 seconds.
👉 The request will be transmitted after 10 seconds, but Plenty is still waiting half a second until resetting the short_calls, and thus we get the "429" error on the first call after the delay.

Because of this, we need to give some buffer time for Plenty before starting, not before stopping.

2 seconds is meant to be "on the safe" side, but if that seems to big, an alternative could be calling `short_seconds_left.ceil` instead of `short_seconds_left.to_i` in order to ensure we always round UP.

@thelastinuit @mwessendorf  care to comment?

Reply:
Oh yeah, i remember this. it seems good to me. @mwessendorf ?